Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs

```diff
@@ -180,6 +184,10 @@ EqualityStrategy IEquivalencyAssertionOptions.GetEqualityStrategy(Type type)
             {
                 strategy = EqualityStrategy.ForceEquals;
             }
+            else if (type.IsRecord())
```

Comment:
Just sharing thoughts.
If any users report a slowdown on `BeEquivalentTo`, caching of `IsRecord` might be a place to look.

Reply:
I still wanted to do a performance test on the entire test suite. Will add it to the 6.0 release list.